hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictorImpl.java

```diff
@@ -38,22 +47,26 @@
 public class EvictorImpl implements Evictor {
```

Comment:
Would it not be an idea to EvictorImpl per RecordStore instead of this stateless object? Then you can move all eviction related logic to this class instead of having part of it in the evictable record store and another part in the evictor.